.github/workflows/c-cpp.yml

```diff
@@ -9,13 +9,98 @@ on:
 jobs:
   build:
 
-    runs-on: ubuntu-latest
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    container: ${{ matrix.config.container }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {name: 'ubuntu-14.04 gcc-4.8', os: ubuntu-latest, container: 'ubuntu:14.04', cc: 'gcc', cxx: 'g++', tag: '4.8', toolchain: 'ppa:ubuntu-toolchain-r/test'}
+          - {name: 'ubuntu-14.04 gcc-4.9', os: ubuntu-latest, container: 'ubuntu:14.04', cc: 'gcc', cxx: 'g++', tag: '4.9', toolchain: 'ppa:ubuntu-toolchain-r/test'}
+          - {name: 'ubtunu-18.04 gcc-5', os: ubuntu-18.04, cc: 'gcc', cxx: 'g++', tag: '5'}
+          - {name: 'ubtunu-18.04 gcc-6', os: ubuntu-18.04, cc: 'gcc', cxx: 'g++', tag: '6'}
+          - {name: 'ubtunu-18.04 gcc-7', os: ubuntu-18.04, cc: 'gcc', cxx: 'g++', tag: '7'}
+          - {name: 'ubtunu-18.04 gcc-8', os: ubuntu-18.04, cc: 'gcc', cxx: 'g++', tag: '8'}
+          - {name: 'ubtunu-20.04 gcc-9', os: ubuntu-20.04, cc: 'gcc', cxx: 'g++', tag: '9'}
+          - {name: 'ubtunu-20.04 gcc-10', os: ubuntu-20.04, cc: 'gcc', cxx: 'g++', tag: '10'}
+          - {name: 'ubtunu-20.04 gcc-11', os: ubuntu-20.04, cc: 'gcc', cxx: 'g++', tag: '11'}
```

Comment:
```suggestion
          - {name: 'ubtunu-22.04 gcc-11', os: ubuntu-22.04, cc: 'gcc', cxx: 'g++', tag: '11'}
```

Reply:
I think maybe there are fixes in the gcc-11 in 22.04 that avoids the warning seen there. We can of course fix and surround in pragmas like the others.. but we could also just suggest using 22.04 if you really need gcc-11.

Reply:
I agree that we should change the gcc-11 test to run on Ubuntu 22.04, but I feel that should be done in your branch that addresses the Ubuntu 22.04 bootstrap issues.

If I change this to Ubuntu 22.04 in this PR, the CI workflow will fail. (That was the first thing I tried when trying to build with gcc-11.)